drivers/flash/nrf_qspi_nor.c

```diff
@@ -1109,7 +1106,9 @@ static int qspi_nor_write_protection_set(const struct device *dev,
  */
 static int qspi_nor_configure(const struct device *dev)
 {
-	const struct qspi_nor_config *params = dev->config;
+	if (!dev) {
+		return -ENXIO;
+	}
```

Comment:
Other drivers don't perform such checks, so perhaps this should be removed completely?

Reply:
I was also considering that or CHECKIF. I will remove it.

Reply:
Done